packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -249,6 +251,22 @@ export class MetaMaskHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  public async ethSwitchChain(chainId: number): Promise<void> {
+    const hexChainId = ethers.utils.hexValue(chainId);
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
```

Comment:
might be useful to create an ethereum provider interface that we can pass in as the generic return type that types everything from the metamask provider (https://github.com/MetaMask/detect-provider/blob/5ce916fc24779c4b36741531a41d9c8b3cbb0a17/src/index.ts#L1) and adds the request func typings: https://docs.metamask.io/guide/ethereum-provider.html#ethereum-request-args

Reply:
Or looks like that ethers.externalProvider would be even better